components/fxa-client/src/ffi.rs

```diff
@@ -125,34 +126,14 @@ impl From<AccessTokenInfo> for AccessTokenInfoC {
     }
 }
 
-#[repr(C)]
-pub struct ProfileC {
-    uid: *mut c_char,
-    email: *mut c_char,
-    avatar: *mut c_char,
-    avatar_default: u8, // JNA dislikes booleans.
-    display_name: *mut c_char,
-}
-
-impl Drop for ProfileC {
-    fn drop(&mut self) {
-        unsafe {
-            destroy_c_string(self.uid);
-            destroy_c_string(self.email);
-            destroy_c_string(self.avatar);
-            destroy_c_string(self.display_name);
-        }
-    }
-}
-
-impl From<Profile> for ProfileC {
-    fn from(profile: Profile) -> Self {
-        ProfileC {
-            uid: rust_string_to_c(profile.uid),
-            email: rust_string_to_c(profile.email),
-            avatar: rust_string_to_c(profile.avatar),
-            avatar_default: profile.avatar_default as u8,
-            display_name: opt_rust_string_to_c(profile.display_name),
+impl From<Profile> for ffi_types::Profile {
+    fn from(p: Profile) -> Self {
+        ffi_types::Profile {
+            avatar: Some(p.avatar),
+            avatar_default: Some(p.avatar_default),
+            display_name: p.display_name,
```

Comment:
why is `display name` treated differently from `avatar`? Is display name the only one we have left that can be `nil`?

Reply:
They should all be optional instead.